sdk/python/kfp/v2/dsl/_pipeline.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Pipeline class and decorator function definition."""
+
+from typing import Any, Callable
+
+from kfp.components import _components
+from kfp.components import _naming
+from kfp.dsl import _container_op
+from kfp.dsl import _ops_group
+from kfp.v2.dsl import component_bridge
+
+# This handler is called whenever the @pipeline decorator is applied.
+# It can be used by command-line DSL compiler to inject code that runs for every
+# pipeline definition.
+_pipeline_decorator_handler = None
+
+
+def pipeline(name: str = None, description: str = None) -> Callable[..., Any]:
```

Comment:
I'm not sure this function needs to be duplicated. It seems to be unused here.

Reply:
Removed.